helm-charts/falcon-sensor/templates/container_secret.yaml

```diff
@@ -1,16 +1,17 @@
 {{- if .Values.container.enabled }}
 {{- if .Values.container.image.pullSecrets.enable }}
+{{- if not .Values.container.image.pullSecrets.name }}
 {{- $registry := .Values.container.image.pullSecrets.registryConfigJSON }}
 apiVersion: v1
 kind: Secret
 metadata:
-  name: {{ include "falcon-sensor.fullname" . }}-pull-secret
+  name: {{ .Values.container.image.pullSecrets.name | default (printf "%s-pull-secret" (include "falcon-sensor.fullname" .)) }}
```

Comment:
I've kept the secret name changed here to keep the consistency (even if not displayed) in case conditions will be changed